docs/adding_endpoints.md

```diff
@@ -0,0 +1,86 @@
+
+```
+
+We can automatically set up endpoints to access *Genes* by adding the following block:
+
+``` yaml
+accessible_via:
+  name: genes
+  description: 'Retrieve gene information. Example: chr = chr1, gene_name = DDX11L1, gene_type = transcribed_unprocessed_pseudogene'
+  filter_by: _id, chr, gene_name, gene_type
+  filter_by_range: start, end
+  return: _id, chr, gene_name, gene_type, start, end
+```
+
+The description of each field follows:
+
+* `name`: defines the prefix of the endpoint name, in this example, all endpoints will start with `GET /genes/`.
+* `description`: OpenAPI description of the endpoint. It will be consumed by OpenAPI clients such as Swagger.
+
+* `filter_by`: a list of searchable properties that can be searched as query parameters. They define the "Filter by a list of fields" endpoint. All parameters must be defined in the `properties` object. Each field is optional, but **at least one field must be specified**. In this example, we can search a gene via: `GET /genes?chr=chr1&gene_name=ACT1&gene_type=pseudogene` for example. `_id` is the only exception as it's a reserved field and will not be read as a query parameter. If `_id` is set in this list, a separate endpoint "Find by ID" will be set, in this example: `GET /genes/{id}` will be created.
+* `filter_by_range`: this list complements `filter_by` by specifying that `start` and `end` will define a range query. They can be used along with the other fields for searching. Currently, only `start` and `end` values are acceptable for this field.
+* `return`: the list of fields that will be returned by the endpoint. Each gene object will contain the fields: `_id, chr, gene_name, gene_type, start, end` in the response object.
+
+So far, we discussed how to enable the endpoints: "Filter by a list of fields" (including range query) and "Find by ID". Now, let's discuss how to enable graph based endpoints automatically.
+
+## Graph based endpoints
+
+Currently, the only graph based endpoints that are supported are the return of a list of children and parents of a node object.
+
+Relationships must be explicitly defined in edge models.
+
+For example:
+
+``` yaml
+caqtl:
+  represented_as: edge
+  label_in_input: caqtl
+  label_as_edge: VARIANT_OPEN_CHROMATIC_REGION
+  db_collection_name: variant_open_chromatic_region_links
+  db_collection_per_chromosome: false
+  relationship:
+    from: sequence variant
+    to: open chromatin region
+  properties:
+    chr: str
+    rsid: str
+```
+
+The `relationship` object can defined in any `edge` object and it must contain two keys: `from` and `to`.
+
+The values of `from` and `to` must be a singular class name of a node also defined in the `schema-config.yaml` file.
+
+In our example above, we are defining a relationship (sequence variant) -> (open chromatin region). This will automatically define the following endpoints, if the objects have an `accessible_via` object in their schema definitions:
+
+* `GET /variants/children`
```

Comment:
Can a node have more than one type of parent and more than one type of child? If so can we filter by type when we query the parent or child?

Reply:
yes, it can have as many as defined by the `relationship` blocks. The endpoint will return all of them.

Reply:
if they have children a `/children` endpoint will be created, similarly with parents

Reply:
I mean if a node have two types of children, can we filter the children by different relationship (edge)? something like /children?edge= topld

Reply:
oh actually... I just realized the endpoint is missing the `id` field... I will update it

oh got it! it's a great suggestion, so far we don't support it, if we have a use case, we can add it, but as of now the endpoint just allows a broad exploration of the data

Reply:
yeah we should add this for ontologies specifically as they have very different meanings, the edges.   There are also properties on edges that could could be filtered but no need to implement that without full use case.